src/wieldmesh.cpp

```diff
@@ -371,19 +374,43 @@ void WieldMeshSceneNode::setItem(const ItemStack &item, Client *client)
 					break;
 				}
 				case NDT_NORMAL:
-				case NDT_ALLFACES: {
+				case NDT_ALLFACES:
+				case NDT_LIQUID:
+				case NDT_FLOWINGLIQUID: {
 					setCube(f, def.wield_scale);
 					break;
 				}
 				default: {
-					MeshMakeData mesh_make_data(client, false);
-					MapNode mesh_make_node(id, 255, 0);
-					mesh_make_data.fillSingleNode(&mesh_make_node);
-					MapBlockMesh mapblock_mesh(&mesh_make_data, v3s16(0, 0, 0));
-					scene::SMesh *mesh = cloneMesh(mapblock_mesh.getMesh());
-					translateMesh(mesh, v3f(-BS, -BS, -BS));
-					postProcessNodeMesh(mesh, f, m_enable_shaders, true,
-						&m_material_type, &m_colors);
+					MeshMakeData mesh_make_data(client, false, false);
+					MeshCollector collector(false);
+					mesh_make_data.setSmoothLighting(false);
+					MapblockMeshGenerator gen(&mesh_make_data, &collector);
+					gen.renderSingle(id);
+					mesh = new scene::SMesh();
+					for (u8 layer = 0; layer < MAX_TILE_LAYERS; layer++)
```

Comment:
if i'm correct you can use a for rangebased loop there

Reply:
No, until I changed C array to std::array =) Hope that won’t break anything =)